processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java

```diff
@@ -1333,6 +1335,42 @@ public void testGroupByWithCardinality()
   }
 
   @Test
+  public void testGroupByWithFirstLast()
```

Comment:
Please also include a test that this works at ingestion time – ie in an incremental index. (some of the use cases from the user lists involved wanting that behavior)